cloudsync/event.py

```diff
@@ -292,27 +299,56 @@ def _fill_event_path(self, event: Event):
                 event.otype = info.otype
                 event.hash = info.hash
     
-    def _filter_event(self, event: Event) -> bool:
-        # event filtering based on root path
-        if not self._root_path:
+    def _filter_event(self, event: Event, from_walk: bool = False) -> bool:
+        # event filtering based on root path and event path
+        # return True = ignore event (filter it out), False = process event
+        if not self._root_path or not self.provider.events_have_path:
```

Comment:
or not event.path

Reply:
Not quite that simple, because of `Event._fill_event_path()` which fills in missing paths from state or the provider itself.

Just tried ripping it out and all tests still pass, do you happen to remember why you added that func ?

If we remove it we should be able to just use `event.path` here.